packages/dnb-eufemia/src/extensions/forms/Value/Upload/Upload.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useMemo } from 'react'
+import classnames from 'classnames'
+import { useValueProps } from '../../hooks'
+import { ValueProps } from '../../types'
+import ValueBlock from '../../ValueBlock'
+import { Anchor } from '../../../../components'
+import Icon from '../../../../components/Icon'
+import Span from '../../../../elements/Span'
+import ListFormat, {
+  ListFormatProps,
+} from '../../../../components/list-format'
+import type {
+  UploadFile,
+  UploadProps,
+} from '../../../../components/upload/types'
+import { fileExtensionImages } from '../../../../components/upload/UploadFileListCell'
+import {
+  BYTES_IN_A_MEGA_BYTE,
+  getFileTypeFromExtension,
+} from '../../../../components/upload/UploadVerify'
+import { format } from '../../../../components/number-format/NumberUtils'
+
+export type Props = ValueProps<Array<UploadFile>> &
+  Omit<ListFormatProps, 'value'> &
+  Pick<UploadProps, 'download'> & {
+    displaySize?: boolean
+  }
+
+function Upload(props: Props) {
+  const {
+    path,
+    value,
+    format,
+    className,
+    variant = 'text',
+    listType,
+    download = false,
+    displaySize = false,
+    ...rest
+  } = useValueProps(props)
+
+  const list = useMemo(() => {
+    const valueToUse =
+      value?.map((uploadFile, index) => {
+        const { file } = uploadFile || {}
+        if (!file) {
+          return
+        }
+        const imageUrl = URL.createObjectURL(file)
+        return (
+          <Span key={index}>
+            {getIcon(file)}
+            <Anchor
+              target="_blank"
+              href={imageUrl}
+              download={download ? file.name : null}
+              rel="noopener noreferrer"
+              className="dnb-anchor--no-launch-icon"
+              left="x-small"
+            >
+              {file.name}
+              {getSize(displaySize, file.size)}
+            </Anchor>
+          </Span>
+        )
+      }) || undefined
+
+    if (typeof valueToUse === 'undefined') {
+      return undefined
+    }
+
+    return (
+      <ListFormat
+        value={valueToUse}
+        format={format}
+        variant={variant}
+        listType={listType}
+      />
+    )
```

Comment:
```suggestion
    if (valueToUse) {
      return (
        <ListFormat
          value={valueToUse}
          format={format}
          variant={variant}
          listType={listType}
        />
      )
    }
```

We except `valueToUse` to be something. So we want to have the "check" related to it.